### PR TITLE
YARN-11070. Minimum resource ratio is overridden by subsequent labels

### PR DESCRIPTION
**Pull request by 9uapaw**
### For code changes:

- [ ] Does the title or this PR starts with the corresponding JIRA issue id (e.g. 'HADOOP-17799. Your PR title ...')?
- [ ] Object storage: have the integration tests been executed and the endpoint declared according to the connector-specific documentation?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the `LICENSE`, `LICENSE-binary`, `NOTICE-binary` files?

### CODE REVIEW

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/ParentQueue.java [hunk 1]
```diff
@@ -105,7 +106,7 @@
   // leaf queue can use this to calculate effective resources.
   // This field will not be edited, reference will point to a new immutable map
```
Comment (brumi1024) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/ParentQueue.java]: AFAIK this isn't true anymore.

File: hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/ParentQueue.java [hunk 2]
```diff
@@ -1328,8 +1329,8 @@ private void calculateEffectiveResourcesAndCapacity(String label,
       }
     }
 
-    effectiveMinRatioPerResource = getEffectiveMinRatioPerResource(
-        configuredMinResources, numeratorForMinRatio);
+    effectiveMinRatioPerResource.put(label, getEffectiveMinRatioPerResource(
```
Comment (brumi1024) [inline hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/capacity/ParentQueue.java]: Nit/idea: I think a name like effectiveMinResourceRatio is a bit more descriptive.

**Review by brumi1024 — COMMENTED** [review]
@9uapaw thanks for the patch, nice find. I had two minor comments.

**Comment by szilard-nemeth** [comment]
Hi @9uapaw ,
Thanks for working on this.
Patch LGTM, committed to trunk.
Thanks @brumi1024for the review.

**Pull request closed without merging**